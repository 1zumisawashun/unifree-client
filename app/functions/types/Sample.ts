// /*
//  * @see https://twitter.com/mattpocockuk/status/1740418559623508269?s=12&t=0Bs_ltBYiO3nhiiL9YZAEw
//  */
// type InferValueFromColor<Color extends string> =
//   Color extends `${infer N}-${infer C}-${infer T}`
//     ? {
//         namespace: N;
//         color: C;
//         tone: T;
//       }
//     : never;

// type Example = InferValueFromColor<"text-red-500">;

// /*
//  * @see https://twitter.com/mattpocockuk/status/1740702711652225157?s=12&t=0Bs_ltBYiO3nhiiL9YZAEw
//  */
// const inferValueFromColor = <
//   N extends string,
//   C extends string,
//   T extends number
// >(
//   colorTag: `${N}-${C}-${T}`
// ) => {
//   const [namespace, color, tone] = colorTag.split("-");
//   return {
//     namespace: namespace as N,
//     color: color as C,
//     tone: Number(tone) as T,
//   };
// };

// const example = inferValueFromColor("text-red-500");

// type A = "foo"; // global scope
// type B = A extends infer C
//   ? C extends "foo"
//     ? true
//     : false // *only* inside this expression, C represents A
//   : never;

// // ちょっとユースケース的に考えにくいかも
// // @see https://twitter.com/mattpocockuk/status/1660954698386403332?s=12&t=0Bs_ltBYiO3nhiiL9YZAEw
// type SomeObject = {
//   a: string;
//   b: string;
// };

// export type Example2 = {
//   [K in keyof SomeObject]: {
//     key: K;
//   };
// }[keyof SomeObject];
