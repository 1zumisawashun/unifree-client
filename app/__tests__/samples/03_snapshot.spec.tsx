import Sample from '@/__tests__/utilities/Sample'
import { render } from '@testing-library/react'

describe('snapshotの単体テスト', () => {
  it('正しくレンダリングされる', async () => {
    const { asFragment } = render(<Sample />)
    expect(asFragment()).toMatchSnapshot()
  })
})
