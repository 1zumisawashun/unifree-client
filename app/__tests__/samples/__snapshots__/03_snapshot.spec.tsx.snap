// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshotの単体テスト 正しくレンダリングされる 1`] = `
<DocumentFragment>
  <h1>
    Hello
  </h1>
</DocumentFragment>
`;
